x/mongo/driver/session/client_session.go

```diff
@@ -262,6 +278,30 @@ func (c *Client) UpdateRecoveryToken(response bson.Raw) {
 	c.RecoveryToken = token.Document()
 }
 
+// UpdateSnapshotTime updates the session's value for the atClusterTime field of ReadConcern.
+func (c *Client) UpdateSnapshotTime(response bsoncore.Document) {
+	if c == nil {
+		return
+	}
+
```

Comment:
There should be a check here for `atClusterTime` in the top level of the response if the command was `distinct`. `find` or `aggregate` replies will embed `atClusterTime` in the `cursor` field.